metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/AutocompleteRequestHandler.java

```diff
@@ -256,9 +263,9 @@ private HighlightBuilder getHighlights(@Nullable String field) {
     return highlightBuilder;
   }
 
-  private List<String> getAutocompleteFields(@Nullable String field) {
+  private List<Pair> getAutocompleteFields(@Nullable String field) {
     if (field != null && !field.isEmpty()) {
-      return ImmutableList.of(field);
+      return ImmutableList.of(Pair.of(field, "2.0"));
```

Comment:
This is hardcoded to 2.0 and looks like it treats URN matches and non-URN matches the same. Previously it weighted URNs higher which probably doesn't make sense for auto-complete purposes, but now the conditional on line 220 is the same for both conditions. So we should probably simplify the conditional and likely remove the boostScore entirely?

Reply:
As I checked, we can't remove the boostscore entirely. We need to have higher boost score for ngram fields for non-urn fields in "multi_match" clause to get the expected result. I would suggest that we can have boostscore for non-urn fields from their respective searchable spec(which I already did) and urn field can have only default boost score, as you have rightly mention that  it doesn't make sense for auto-complete purposes.

Please find the expected multi_match clause

` {
      "multi_match": {
        "query": "aucdl",
        "fields": [
          "name.delimited^1.0",
          "name.ngram^10.0",
          "name.ngram._2gram^10.0",
          "name.ngram._3gram^10.0",
          "name.ngram._4gram^10.0",
          "urn.delimited^1.0",
          "urn.ngram^1.0",
          "urn.ngram._2gram^1.0",
          "urn.ngram._3gram^1.0",
          "urn.ngram._4gram^1.0"
        ],
        "type": "phrase",
        "operator": "OR",
        "slop": 0,
        "prefix_length": 0,
        "max_expansions": 50,
        "zero_terms_query": "NONE",
        "auto_generate_synonyms_phrase_query": true,
        "fuzzy_transpositions": true,
        "boost": 1.0
      }
    }`

In case of function at line 266, we can put condition to have default boostscore of 10.0 for non-urn field and boostscore of 1.0 for urn field.

Reply:
Ok, there are two things to consider here, the relative weights between the fields being used for the query AND the relative weights of the subfields (delimited, ngram, etc). Most likely we should be applying a multiplier. The URN while important as a unique identifier for general search is not as important here so the urn field weight is likely 1.0.

Next lets consider a multiplier for ngram. I am not seeing these being applied -> https://github.com/datahub-project/datahub/blob/b74ba11d937ef4b902f320e1bdb39a7ece35ffc4/metadata-service/configuration/src/main/resources/application.yaml#L240

The field weights for ngrams should be impacted by that configuration and it doesn't appear to be based on. the example query above. 

Generally the weight formula for ngram subfields should be something like:
`<field annotation weight>` * `<configuration ngram factor>`

Reply:
Hi @david-leifker, As suggested done the following changes

- add "match" clause for exact matches
- make URN field default boostscore to 1.0
- make ngram subfields weight as per the formula suggested
- add a testcase in class "SampleDataFixtureTestBase.java"
- add testdata in containerindex_v2.json.gz
- reverted to "bool_prefix" type in "multi_match" clause

Please find the final query

` {
      "multi_match": {
        "query": "container",
        "fields": [
          "name.delimited^1.0",
          "name.ngram^10.0",
          "name.ngram._2gram^12.0",
          "name.ngram._3gram^15.0",
          "name.ngram._4gram^18.0",
          "urn.delimited^1.0",
          "urn.ngram^1.0",
          "urn.ngram._2gram^1.0",
          "urn.ngram._3gram^1.0",
          "urn.ngram._4gram^1.0"
        ],
        "type": "bool_prefix",
        "operator": "OR",
        "slop": 0,
        "prefix_length": 0,
        "max_expansions": 50,
        "zero_terms_query": "NONE",
        "auto_generate_synonyms_phrase_query": true,
        "fuzzy_transpositions": true,
        "boost": 1
      }
    },
    {
        "match": {
        "name.keyword": {
          "query": "container",
          "operator": "OR",
          "prefix_length": 0,
          "max_expansions": 50,
          "fuzzy_transpositions": true,
          "lenient": false,
          "zero_terms_query": "NONE",
          "auto_generate_synonyms_phrase_query": true,
          "boost": 10
        }
      }
       }`